### PR TITLE
Issue #8533: add to top button

### LINKED ISSUE #8533: To Top Button on website
As mentioned in #8244 a button which scrolls to the top of the page would be nice.

### PR DESCRIPTION
**Pull request by jo2**
fix #8533

[preview](https://jo2.github.io/checkstyle-top-button-right/)

I created this PR anew because there were some issues with my git history.

### CODE REVIEW

**Comment by strkkk** [comment]
@jo2 there is failure because some file is missing line feed
from teamcity:
```
src/site/resources/js
checkstyle.js (7)
3:  JSHint: 'let' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz). (W104)
4:  JSHint: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz). (W104)
6:  JSHint: 'arrow function syntax (=&gt;)' is only available in ES6 (use 'esversion: 6'). (W119)
7:  JSHint: Missing &quot;use strict&quot; statement. (E007)
11:  JSHint: 'arrow function syntax (=&gt;)' is only available in ES6 (use 'esversion: 6'). (W119)
17:  JSHint: 'arrow function syntax (=&gt;)' is only available in ES6 (use 'esversion: 6'). (W119)
18:  JSHint: Missing &quot;use strict&quot; statement. (E007)
```
and 
```
src/site/resources/js
checkstyle.js (7)
2:  JSLint: /*global*/ requires the Assume a browser option.
3:  JSLint: Unexpected ES6 feature 'let'.
4:  JSLint: Unexpected ES6 feature 'const'.
6:  JSLint: Unexpected ES6 feature '=&gt;'.
11:  JSLint: Unexpected ES6 feature '=&gt;'.
11:  JSLint: Unused 'event'.
17:  JSLint: Unexpected ES6 feature '=&gt;'.
```

**Comment by romani** [comment]
From Appveyor:
```
[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\xdocs\google_style.xml:32: Old site links should not be used, please use https://checkstyle.org [noSourceforgeIoLinks]
[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\xdocs\sun_style.xml:32: Old site links should not be used, please use https://checkstyle.org [noSourceforgeIoLinks
```

Please update line number at 
https://github.com/checkstyle/checkstyle/blob/master/config/suppressions.xml#L106 to 32.

**Comment by jo2** [comment]
@romani  I didn't change those lines in `google_style.xml` and `sun_style.xml` so I think that problems originates from somewhere else. I can fix it for this PR if you'd like, though.

**Comment by romani** [comment]
please fix in this PR in same commit.
It is caused by adding new line to such XML and we unfortunately has suppression of violation by line number (very fragile model, but we have no better way for now).
You commit changed line numbers - so this file also need to be updated.

**Comment by jo2** [comment]
Done. Now there are stil some city view issues which I can't access.

**Comment by strkkk** [comment]
@jo2 
```
src/site/resources/js
checkstyle.js (2)
7:  JSHint: Missing &quot;use strict&quot; statement. (E007)
18:  JSHint: Missing &quot;use strict&quot; statement. (E007)
```
and
```
src/site/resources/js
checkstyle.js (7)
2:  JSLint: /*global*/ requires the Assume a browser option.
3:  JSLint: Unexpected ES6 feature 'let'.
4:  JSLint: Unexpected ES6 feature 'const'.
6:  JSLint: Unexpected ES6 feature '=&gt;'.
11:  JSLint: Unexpected ES6 feature '=&gt;'.
11:  JSLint: Unused 'event'.
17:  JSLint: Unexpected ES6 feature '=&gt;'.
```

**Comment by romani** [comment]
We never turned JS rules in TC. I am not sure if they valid.

**Comment by jo2** [comment]
I've never worked with Team City so far, so it's not that easy for me to fix these issues. Right now, it only seems to be a problem with es6. I'd really like to use it because it makes the code more readable but if we can't configure the linter accordingly I can change it to es5 which should be accepted.

**Comment by romani** [comment]
We can configure Linter, and I can disable some rules it.
I am ok to not support old standard of JS, most of our users are advanced engineers, they do have modern JS.

**Comment by romani** [comment]
@jo2 ,

I have problem to disable violation on `=>`
from http://www.jslint.com/help.html#function
> Finally, ES6 provides an even shorter form of function expression that leaves out the words function and return:
    (parameters) => expression 
JSLint requires the parens around the parameters, and forbids a { left brace after the => fart to avoid syntactic ambiguity.

Do you know why `{` is not good after "=>" and how to write in better way ?

**Comment by romani** [comment]
I pushed massive suppression, I do not like doing this, but I have no time to fight with JS lint/hint now.

**Comment by jo2** [comment]
@romani I don't understand what is meant with 'no { after =>' because the curly bracket is needed if there is more than one statement. We could write a proxy function encapsulating the other function calls so we can remove the curly braces but that is not a good way to solve this imho. I think the easiest way is to ditch the arrow functions and use `function()` instead. It is not that great but I think it should fix the remaining issues.

**Comment by romani** [comment]
It might be better to follow jlint demands, we do not have that much JS code. Ones we be ready to setup such linters properly, we can update to more modern js

**Comment by romani** [comment]
I pushed one more change, my local IDEA is ok with this code. Lets see how it works on remote.
If still problem, I will exclude this JS from validation, we are inexperienced in JS to have this level of quality.

**Comment by jo2** [comment]
@romani I'll have that in mind for the next pull requests.

**Comment by romani** [comment]
@jo2 , teamcity is passed, please squash all commits in one, regenerate website and lets prepare for final review and merge.

**Comment by jo2** [comment]
I've squashed the commits. The preview is [here](https://jo2.github.io/checkstyle-top-button-right/)

**Comment by romani** [comment]
@rnveach, I see all other are done also, please repeat what you think should be improved

File: src/site/resources/js/checkstyle.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*jslint browser: true */
+/*jslint es6: true */
+/*jshint esversion: 6 */
+/*global window */
+let scrollButton;
+const scrollDistanceToButtonVisibility = 500;
+
+window.addEventListener("load", function () {
+    "use strict";
```
Comment (strkkk) [inline src/site/resources/js/checkstyle.js:9]: I thought this can be added to top of the script instead of adding it to every function

Comment (romani) [reply src/site/resources/js/checkstyle.js:9]: it does not work at most top for some reason.

**Comment by romani** [comment]
I merged PR as release will happen tomorrow
We can improve later on if required.

**Pull request merged**